skbio/sequence/_sequence.py

```diff
@@ -1843,3 +1945,272 @@ def _byte_ownership(self):
+
+        Returns the number of lines the sequence will occupy, the number of
+        sequence characters displayed on each line, and the column width
+        necessary to display position info using the optimal number of sequence
+        chunks.
+
+        """
+        # strategy: use an iterative approach to find the optimal number of
+        # sequence chunks per line. start with a single chunk and increase
+        # until the max line width is exceeded. when this happens, the previous
+        # number of chunks is optimal
+        num_lines = 0
+        num_chars = 0
+        column_width = 0
+
+        num_chunks = 1
+        not_exceeded = True
+        while not_exceeded:
+            line_len, new_chunk_info = self._compute_chunked_seq_line_len(
+                num_chunks)
+            not_exceeded = line_len <= self._width
+            if not_exceeded:
+                num_lines, num_chars, column_width = new_chunk_info
+                num_chunks += 1
+        return num_lines, num_chars, column_width
+
+    def _compute_chunked_seq_line_len(self, num_chunks):
+        """Compute line length based on a number of chunks."""
+        num_chars = num_chunks * self._chunk_size
+
+        # ceil to account for partial line
+        num_lines = int(math.ceil(len(self._seq) / num_chars))
+
+        # position column width is fixed width, based on the number of
+        # characters necessary to display the position of the final line (all
+        # previous positions will be left justified using this width)
+        column_width = len('%d ' % ((num_lines - 1) * num_chars))
+
+        # column width + number of sequence characters + spaces between chunks
+        line_len = column_width + num_chars + (num_chunks - 1)
+        return line_len, (num_lines, num_chars, column_width)
+
+    def _format_chunked_seq(self, line_idxs, num_chars, column_width):
+        """Format specified lines of chunked sequence data."""
+        lines = []
+        for line_idx in line_idxs:
+            seq_idx = line_idx * num_chars
+            chars = str(self._seq[seq_idx:seq_idx+num_chars])
+            chunked_chars = chunk_str(chars, self._chunk_size, ' ')
+            lines.append(('%d' % seq_idx).ljust(column_width) + chunked_chars)
+        return lines
+
+
+class _ElasticLines(object):
+    """Store blocks of content separated by dashed lines.
+
+    Each dashed line (separator) is as long as the longest content
+    (non-separator) line.
+
+    """
+
+    def __init__(self):
+        self._lines = []
+        self._separator_idxs = []
+        self._max_line_len = -1
+
+    def add_line(self, line):
+        line_len = len(line)
+        if line_len > self._max_line_len:
+            self._max_line_len = line_len
+        self._lines.append(line)
+
+    def add_lines(self, lines):
+        for line in lines:
+            self.add_line(line)
+
+    def add_separator(self):
```

Comment:
If a dev concludes symmetry and tries to use `add_veritcal_separator`, it will immediately fail. Meanwhile, `add_horizontal_separator` is perfectly clear from the name. In it's current state I had to look at the source to be sure it was a horizontal separator. It's a minor point though; I'm fine with the current name.